packages/server/src/api/notification/notification-alert.controller.ts

```diff
@@ -0,0 +1,314 @@
+import _ from 'lodash';
+import { Response } from 'express';
+import moment from 'moment-timezone';
+import {
+  ApiRequest,
+  OrgSettingParams,
+} from '../api.router';
+import { AccessRequest, AccessRequestStatus } from '../access-request/access-request.model';
+import { Observation } from '../observation/observation.model';
+import { Unit } from '../unit/unit.model';
+import { UserNotificationSetting } from './user-notification-setting.model';
+import { musterComplianceStatsByDateRange } from '../../util/muster-utils';
+
+type AlertData = { [key: string]: number };
+type OutputRecord = {
+  userSetting: UserNotificationSetting;
+  alertData: AlertData;
+};
+
+class NotificationAlertController {
+
+  // return the number of users per unit requesting medical attention
+  async getUsersRequestingMedicalAttentionByUnit(req: ApiRequest<OrgSettingParams>, res: Response) {
+    const output: OutputRecord[] = [];
+
+    // first get all the users from the org who are registered for the given alert
+    const userSettings = (await getUsersRegisteredForNotification(req.appOrg!.id, 'speakToSomeoneAlert'))
+      .filter((setting: UserNotificationSetting) => {
+        return isEligibleForNotification(setting);
+      });
+
+    // for each user we need to see if it is time to fire an alert and calculate the values
+    for (const setting of userSettings) {
+      const alertData: AlertData = {};
+      const minutes = getMinutesSinceLastNotification(setting);
+      const rows = await Observation.createQueryBuilder('observation')
+        .select('COUNT(observation.unit) as count, observation.unit')
```

Comment:
One thing that is a bit weird about the observation table is that `observation.unit` is actually a user-reported field.  The real unit should actually be acquired by joining the observation table with roster history on `roster_history_entry_id` and fetching the unit ID from there.  I'm not sure which one this notification is supposed to be based on, but I would expect it to be the one from roster history so that it matches the target user's permissions.  I am guessing the final result also needs to be filtered by what units the user has access to.

Reply:
I am a little lost on the permissions comment, my understanding is these endpoints are only ever called from the notification engine.

Reply:
Also regarding joining with roster history - doesn't it actually go a bit further? The alert data is a dictionary of units (names in this case) and the counts. So I would have to join with unit also to actually get the name. Should this be two separate queries where I join with roster entry id and then back fill the names by looking up the resulting IDs?